examples/interactive/cli-platform/File.roc

```diff
@@ -0,0 +1,123 @@
+
+## Write bytes to a file.
+##
+##     # Writes the bytes 1, 2, 3 to the file `myfile.dat`.
+##     File.writeBytes (Path.fromStr "myfile.dat") [1, 2, 3]
+##
+## This opens the file first and closes it after writing to it.
+##
+## To format data before writing it to a file, you can use [File.write] instead.
+writeBytes : Path, List U8 -> Task {} [FileWriteErr Path WriteErr]* [Write [File]*]*
+writeBytes = \path, bytes ->
+    InternalPath.toBytes path
+    |> Effect.fileWriteBytes bytes
+    |> InternalTask.fromEffect
+    |> Task.mapFail \err -> FileWriteErr path err
+
+## Write a [Str] to a file, encoded as [UTF-8](https://en.wikipedia.org/wiki/UTF-8).
+##
+##     # Writes "Hello!" encoded as UTF-8 to the file `myfile.txt`.
+##     File.writeUtf8 (Path.fromStr "myfile.txt") "Hello!"
+##
+## This opens the file first and closes it after writing to it.
+##
+## To write unformatted bytes to a file, you can use [File.writeBytes] instead.
+writeUtf8 : Path, Str -> Task {} [FileWriteErr Path WriteErr]* [Write [File]*]*
+writeUtf8 = \path, str ->
+    InternalPath.toBytes path
+    |> Effect.fileWriteUtf8 str
+    |> InternalTask.fromEffect
+    |> Task.mapFail \err -> FileWriteErr path err
+
+## Read all the bytes in a file.
+##
+##     # Read all the bytes in `myfile.txt`.
+##     File.readBytes (Path.fromStr "myfile.txt")
+##
+## This opens the file first and closes it after reading its contents.
+##
+## To read and decode data from a file, you can use `File.read` instead.
+readBytes : Path -> Task (List U8) [FileReadErr Path ReadErr]* [Read [File]*]*
+readBytes = \path ->
+    InternalPath.toBytes path
+    |> Effect.fileReadBytes
+    |> InternalTask.fromEffect
+    |> Task.mapFail \err -> FileReadErr path err
+
+## Read a [Str] from a file containing [UTF-8](https://en.wikipedia.org/wiki/UTF-8)-encoded text.
+##
+##     # Reads UTF-8 encoded text into a `Str` from the file `myfile.txt`.
+##     File.readUtf8 (Path.fromStr "myfile.txt")
+##
+## This opens the file first and closes it after writing to it.
+## The task will fail with `FileReadUtf8Err` if the given file contains invalid UTF-8.
+##
+## To read unformatted bytes from a file, you can use [File.readBytes] instead.
+readUtf8 :
+    Path
+    -> Task
+        Str
+        [FileReadErr Path ReadErr, FileReadUtf8Err Path _]*
+        [Read [File]*]*
+readUtf8 = \path ->
+    effect = Effect.map (Effect.fileReadBytes (InternalPath.toBytes path)) \result ->
+        when result is
+            Ok bytes ->
+                Str.fromUtf8 bytes
+                |> Result.mapErr \err -> FileReadUtf8Err path err
+
+            Err readErr -> Err (FileReadErr path readErr)
+
+    InternalTask.fromEffect effect
+
+# read :
+#     Path,
+#     fmt
+#     -> Task
+#         Str
+#         [FileReadErr Path ReadErr, FileReadDecodeErr Path [Leftover (List U8)]Decode.DecodeError ]*
+#         [Read [File]*]*
+#     | val has Decode.Decoding, fmt has Decode.DecoderFormatting
+# read = \path, fmt ->
```

Comment:
Is there some difficulty with `File.read` that we're deferring to another PR?

Reply:
Yeah, there are some compiler issues blocking it from working right now.